packages/ckeditor5-find-and-replace/src/findandreplaceui.js

```diff
@@ -140,8 +140,8 @@ export default class FindAndReplaceUI extends Plugin {
 		} );
 
 		this.bind( 'highlightOffset' ).to( state, 'highlightedResult', highlightedResult => {
-			if ( !highlightedResult ) {
-				return null;
+			if ( !highlightedResult && state.results.length === 0 ) {
+				return 0;
```

Comment:
These proposed 2 lines, although it gives expecting results for the end user, are wrong from the logic perspective.

`highlightOffset` is a 0-based index of currently highlighted item. If you assign 0 to it when there are no results or there is no highlighted result you're acting as if there was at least one search result available and the first search result was highlighted. That is not true.

The easiest fix would be to fix it by binding form's `isCounterHidden` observable to state's `searchText` value. If state has non-empty value in `searchText` it means that searching was executed and we can show the counter.

However, the more proper solution is more complex: as I started to look through the code I understood that [I initially messed the `isSearching` observable in the UI plugin](https://github.com/ckeditor/ckeditor5/blob/85f658bf3471c609fbd8644d6a25a425804fb573/packages/ckeditor5-find-and-replace/src/findandreplaceui.js#L103-L116). What I implemented there should be called `isDirty` property - as it tells whether the state differs from what user entered into the UI. `isSearching` should be much simpler. It should be pretty much what I wrote above.

The last idea is the proper fix for it and we should take the occasion to do it as it will make the plugin more intuitive.

Reply:
As per your suggestion, the `isDirty` property has been introduced now. Also, a lot of changes to the code to improve code readability, usability etc.

Reply:
That's right we introduced the `isDirty` property but the change raised by me in this comment was not reverted. It should be reverted because otherwise offset can have misleading value.